### PR TITLE
Rewrite log & blooms.

### PR DESCRIPTION
**Pull request by kaikai1024**
### Description of the Change

* Refactor the filter type used by chain
* Rename log_entry to log
* Chore about log
* Add unit tests for log

### CODE REVIEW

File: cita-chain/types/src/header.rs [hunk 1]
```diff
@@ -33,8 +33,6 @@ pub use crate::block_number::BlockNumber;
 use hashable::{Hashable, HASH_NULL_RLP};
 use util::HeapSizeOf;
 
-// from parity
-pub use crate::log_entry::LogBloom;
 lazy_static! {
     pub static ref ZERO_LOGBLOOM: LogBloom = LogBloom::from([0x00; 256]);
 }
```
Comment (ouwenkg) [inline cita-chain/types/src/header.rs]: `Line 39` has GPL suspicion.

File: cita-chain/types/src/log.rs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        })
+    }
+}
+
+impl HeapSizeOf for Log {
+    fn heap_size_of_children(&self) -> usize {
+        self.topics.heap_size_of_children() + self.data.heap_size_of_children()
+    }
+}
+
```
Comment (ouwenkg) [inline cita-chain/types/src/log.rs]: `HeapSizeOf` is useless now.

File: cita-chain/types/src/header.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-chain/types/src/header.rs]: ok

File: cita-chain/types/src/log.rs (hunk 2 above)
Comment (kaikai1024) [reply cita-chain/types/src/log.rs]: ok, I'll have a try

**Pull request merged**